read_buffer/src/chunk.rs

```diff
@@ -205,17 +205,20 @@ impl Chunk {
         predicate: Predicate,
         group_columns: &Selection<'_>,
         aggregates: &[(ColumnName<'_>, AggregateType)],
-    ) -> Option<table::ReadAggregateResults> {
+    ) -> Result<table::ReadAggregateResults> {
         // read lock on chunk.
         let chunk_data = self.chunk_data.read().unwrap();
 
-        // Lookup table by name and dispatch execution.
-        //
-        // TODO(edd): this should return an error
-        chunk_data
+        let table = chunk_data
             .data
             .get(table_name)
-            .map(|table| table.read_aggregate(predicate, group_columns, aggregates))
+            .ok_or(Error::TableNotFound {
+                table_name: table_name.to_owned(),
+            })?;
+
```

Comment:
I think you can use the `OptionExt` trait to make this:

```suggestion
            .context(TableNotFound {table_name})?;

```

Reply:
Fixed this in a couple of places in 585213e. Thanks.